go/vt/vtorc/logic/orchestrator.go

```diff
@@ -267,12 +266,14 @@ func DiscoverInstance(instanceKey inst.InstanceKey, forceDiscovery bool) {
 		InstanceLatency: instanceLatency,
 		Err:             nil,
 	})
+	// we turn on HitAtLeastOneDiscovery first time
+	// con: this will result in extra memory hit for every recovery cycle.
+	process.HitAtLeastOneDiscovery.CompareAndSwap(false, true)
```

Comment:
only after first successful discovery. Should we instead make it only after first discoveryInstance call regardless of success or failure? 
I though successful is better.